packages/asset-service/src/generateAssetData/generateAssetData.ts

```diff
@@ -58,15 +71,22 @@ const generateAssetData = async () => {
               return { ...uniqueTokens[key], icon }
             }
           })
+
           modifiedTokens = modifiedTokens.concat(newModifiedTokens)
         }
+
         const baseAssetWithTokens: BaseAsset = {
           ...baseAsset,
           tokens: modifiedTokens
         }
+
         return baseAssetWithTokens
       } else {
-        return baseAsset
+        if (!blacklistedAssets.includes(baseAsset.caip19)) {
+          return baseAsset
+        }
+
+        return
```

Comment:
unhappy path first 🙏🏽 
```suggestion
        if (blacklistedAssets.includes(baseAsset.caip19)) { return }

        return baseAsset
```

Reply:
Will be removed after the first review is completed anyway